hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonFileInputFormat.java

```diff
@@ -145,9 +158,33 @@ public CarbonTable getOrCreateCarbonTable(Configuration configuration) throws IO
           externalTableSegments.add(seg);
         }
       }
-      // do block filtering and get split
-      List<InputSplit> splits =
-          getSplits(job, filter, externalTableSegments, null, partitionInfo, null);
+      List<InputSplit> splits = new ArrayList<>();
+      if (isSDK) {
+        for (CarbonFile carbonFile : getAllCarbonDataFiles(carbonTable.getTablePath())) {
```

Comment:
move it to method getAllFileSplits(), and called it from CarbonReaderBuilder.build() as discussed above

Reply:
ok